ckanext/csrf_filter/anti_csrf.py

```diff
@@ -367,7 +367,16 @@ def apply_token(response, request=None):
     """ Rewrite HTML to insert tokens if applicable.
     If a new token is generated, it will be added to 'response' as a cookie.
     """
-    html = getattr(response, 'data', None)
+    try:
+        if not response.charset:
+            # Provide default charset
+            response.charset = 'utf-8'
```

Comment:
`response.get_data` tries to decode the data on the response using the response's charset... we've found instances where the response's is None, so setting a default here.